Gemfile

```diff
@@ -85,7 +85,7 @@ gem "inifile",                        "~>3.0",     :require => false
 gem "logging",                        "~>1.8",     :require => false  # Ziya depends on this
 gem "net_app_manageability",          ">=0.1.0",   :require => false
 gem "net-ping",                       "~>1.7.4",   :require => false
-gem "net-ssh",                        "~>2.9.2",   :require => false
+gem "net-ssh",                        "= 4.0.0.alpha4",   :require => false
```

Comment:
I agree, from past upgrades of this gem, the underlying code that consumes it is hard to test as we can't easily run it on travis because we need public hosts of various STATIC configurations to assert against.    Perhaps, something like VCR could be used in the future to record these interactions.

I'm not sure about the automate state machines but the primary remaining user is the `Host#connect_ssh` method which is used in several places including `Host#scan_from_queue`, `Host#verify_credentials_with_ssh` and `Host#ssh_run_script` which is exposed in the MiqAeServiceHost which I believe exposes it to automate script writers.  The test surface is possible to manually test but we'd need to do that before we can upgrade the gem.

A backport of the PR to a stable branch, while better, would also require some manual testing.